packages/guilded.js/lib/gateway/handler/CalendarEventHandler.ts

```diff
@@ -5,17 +5,28 @@ import { GatewayEventHandler } from "./GatewayEventHandler";
 
 export class CalendarEventHandler extends GatewayEventHandler {
     calendarEventCreated(data: WSCalendarEventCreated): boolean {
+        const existingCalendar = this.client.calendars.cache.get(data.d.calendarEvent.id);
+        if (existingCalendar) return this.client.emit(constants.clientEvents.CALENDAR_EVENT_CREATED, existingCalendar);
+        
         const newCalendarEvent = new CalendarEvent(this.client, data.d.calendarEvent);
+        if (this.client.calendars.shouldCacheCalendar) this.client.calendars.cache.set(newCalendarEvent.id, newCalendarEvent);
         return this.client.emit(constants.clientEvents.CALENDAR_EVENT_CREATED, newCalendarEvent);
     }
 
     calendarEventUpdated(data: WSCalendarEventUpdated): boolean {
+        const existingCalendar = this.client.calendars.cache.get(data.d.calendarEvent.id);
+        const oldCalendar = existingCalendar?._clone();
+        const updatedCalendar = existingCalendar?._update(data.d.calendarEvent);
         const newCalendarEvent = new CalendarEvent(this.client, data.d.calendarEvent);
-        return this.client.emit(constants.clientEvents.CALENDAR_EVENT_UPDATED, newCalendarEvent);
+        if (this.client.calendars.shouldCacheCalendar) this.client.calendars.cache.set(newCalendarEvent.id, newCalendarEvent);
+        return this.client.emit(constants.clientEvents.CALENDAR_EVENT_UPDATED, newCalendarEvent, oldCalendar ?? null);
     }
 
     calendarEventDeleted(data: WSCalendarEventDeleted): boolean {
+        const existingCalendar = this.client.calendars.cache.get(data.d.calendarEvent.id);
+        const deleteCalendar = existingCalendar?._update(data.d.calendarEvent);
         const newCalendarEvent = new CalendarEvent(this.client, data.d.calendarEvent);
+        if (this.client.calendars.shouldCacheCalendar) this.client.calendars.cache.delete(newCalendarEvent.id);
```

Comment:
this should use a new cache option (e.g. `removeCalendarEventOnDelete`) like we do for others, in case people want to keep it cached even after deletion